File: Linguagems de Programação 2/C# Aula 6 - Herança/Aula6/Aula6/Aluno.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Aula6
{
    class Aluno : Pessoa
    {
        int matricula;
        float notaP1;
        float notaP2;
        int faltas;
        int limiteDeFaltas = 10;
        float mediaDeAprovação = 7.0f;
        float mediaDeReprovação = 3.0f;

        public int Matricula { get => matricula; private set => matricula = value; }
        public float NotaP1 { get => notaP1; private set => notaP1 = value; }
        public float NotaP2 { get => notaP2; private set => notaP2 = value; }
        public int Faltas { get => faltas; private set => faltas = value; }
        public int LimiteDeFaltas { get => limiteDeFaltas; private set => limiteDeFaltas = value; }
        public float MediaDeAprovação { get => mediaDeAprovação; private set => mediaDeAprovação = value; }
        public float MediaDeReprovação { get => mediaDeReprovação; private set => mediaDeReprovação = value; }

        public float Media() { return (NotaP1 + NotaP2) / 2; }

        public void ImprimeFaltas() => Console.WriteLine(faltas);

        public string Resultado()
        {
            if (Media() >= MediaDeAprovação) return "Aprovado";
            if (Media() < MediaDeReprovação) return "Reprovado";
            else return "Reavaliação";
        } 

        public Aluno(int matricula, int notaP1, int notaP2, int faltas, string nome, string endereço, string rg, string cpf) : base (nome, endereço, rg, cpf)
        {
            Matricula = matricula;
            NotaP1 = notaP1;
            NotaP2 = notaP2;
            Faltas = faltas;
        }

        public Aluno(){}
    }
}

File: AED 3/Aula 01/Aluno.java
package aula0602;
public class aluno {
String nome;
int codmat;
String cpf;
char sexo;
double nota_P1;
double nota_P2;
double nota_P3;
double media = 0;
double maior1;
double maior2;
public aluno(String nome, int codmat, String cpf, char sexo, double nota_P1, double nota_P2, double nota_P3) {
this.nome = nome;
this.codmat = codmat;
this.cpf = cpf;
this.sexo = sexo;
this.nota_P1 = nota_P1;
this.nota_P2 = nota_P2;
this.nota_P3 = nota_P3;
}
public void imprime_Aluno() {
System.out.println(this.nome);
System.out.println(this.codmat);
System.out.println(this.cpf);
System.out.println(this.sexo); 
}
public void imprime_Sexo() {
System.out.println(this.nome);
if(sexo == 'M') {
System.out.println("Masculino");
}else if (sexo == 'F'){
System.out.println("Feminino");
}
}
public double media_Aluno() {
maior1 = nota_P1;
maior2 = nota_P2;
if(nota_P3 > nota_P1) maior1 = nota_P3;
else if (nota_P3 > nota_P2) maior2 = nota_P3; 
if(nota_P1 > maior2) maior2 = nota_P1;
media = ((maior1 + maior2) / 2);
return media;
}
public void resultado() {
if(media >= 6) {
System.out.println("Aprovado");
}else {
System.out.println("Reprovado");
}
}
}
